packages/react/dialog/src/Dialog.tsx

```diff
@@ -347,6 +339,19 @@ const DialogContentImpl = React.forwardRef((props, forwardedRef) => {
           aria-label={ariaLabel || undefined}
           data-state={getState(context.open)}
           {...contentProps}
+          onEscapeKeyDown={composeEventHandlers(props.onEscapeKeyDown, () => {
+            preventAutoFocusRef.current = false;
+          })}
+          onPointerDownOutside={composeEventHandlers(
+            props.onPointerDownOutside,
+            (event) => {
+              const originalEvent = event.detail.originalEvent;
+              const ctrlLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === true;
+              const isRightClick = originalEvent.button === 2 || ctrlLeftClick;
+              preventAutoFocusRef.current = isRightClick || !props.disableOutsidePointerEvents;
```

Comment:
Rather than focusing on a left pointer down I've instead excluded the auto focus when right click or pointer events enabled (e.g. non modal).

This was the cleanest way I could see to handle the differing behaviour between the modality modes, let me know what you think though.

Reply:
I feel like I must be missing something here because my immediate thought is to wonder why we can't have the separate implementations in the different modal type components... like:

```js
const DialogContentModal = () => {
	// ...
	return (
		<DialogContentImpl 
			disableOutsidePointerEvents
			onUnmountAutoFocus={composeEventHandlers(onCloseAutoFocus, (event) => {
          		event.preventDefault();
          		if (!preventAutoFocusRef.current) context.triggerRef.current?.focus();
        	})}
			onPointerDownOutside={event => {
				const originalEvent = event.detail.originalEvent;
				const ctrlLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === true;
				const isRightClick = originalEvent.button === 2 || ctrlLeftClick;
				preventAutoFocusRef.current = isRightClick
			}}
		/>
	);
}

// Doesn't need the trigger focus because pointer events aren't disabled?
const DialogContentNonModal = () => {
	// ...
	return (
		<DialogContentImpl 
			disableOutsidePointerEvents
			onUnmountAutoFocus={composeEventHandlers(onCloseAutoFocus, (event) => {
				event.preventDefault();
			})}
		/>
	);
}

Reply:
Although, I'm curious why trigger focus doesn't happen when `disableOutsidePointerEvents` is `false` in your logic here? I think the original ticket was to specifically explicitly focus the trigger but this looks like it wouldn't get explicitly focused now in a non-modal dialog?

Reply:
Yeah we can totally do this though I stopped short because we can't do it for `DropdownMenu` unless we move the trigger to `Menu`. I suppose we could have the implementation be different for that case.. or even bite the bullet and move the trigger?